dist/configure

```diff
@@ -144,36 +150,140 @@ int main () {
   return 0;
 }
 EOF
-  $CC -Werror -c $file -o /dev/null
+  $CC $CROSS_CFLAGS -Werror -c $file -o /dev/null
+}
+
+# Usage info
+show_help () {
+  printf "\nUsage: configure -target <triple>\n\n"
+  printf "This script configures HACL/Evercrypt. You can specify the following options:\n\n"
+  printf "    -target         Specify the target triple for the build. This follows the\n"
+  printf "                    Clang target triple convention.\n"
+  printf "                    Details: https://clang.llvm.org/docs/CrossCompilation.html\n"
+  printf "                    Currently supported triples are:\n"
+  printf "                    * aarch64-none-linux-android\n"
+  printf "                    * aarch64-none-linux-gnu\n"
+  printf "    --disable-bzero Do not use explicit_bzero (binary will work with an old GLIBC)\n"
+  printf "    --disable-ocaml Disable OCAML bindings\n"
+  printf "\n"
 }
 
 # Main auto-detection
 
 echo -n > Makefile.config
 echo -n > config.h
 
+# Default arguments
+disable_ocaml=0
+disable_bzero=0
+
+# Parse command line arguments.
+all_args=("$@")
+while [ $# -gt 0 ]; do
+    case "$1" in
+        -target) build_target="$2"; shift ;;
+        --disable-ocaml) disable_ocaml=1 ;;
+        --disable-bzero) disable_bzero=1 ;;
+        --help) show_help; exit 0 ;;
+        *) show_help; exit 2 ;;
+    esac
+    shift
+done
+
+# We need to have the following to decide what to build and what not.
+# * build_target: the llvm target triplet, e.g. armv7a-none-eabi
+# This is being decomposed into the following information.
+# * target_arch: the target architecture, e.g. aarch64, arm, x86_64, i386
+# * target_sys: the target system, e.g. none, linux, win32, darwin
+# * target_abi: the target abi, e.g. eabi, gnu, android, macho
+
+if [ ! -z "$build_target" ]; then
+  echo "Doing cross build for $build_target. Make sure that your toolchain is configured properly."
+  if [[ "$build_target" == "aarch64-none-linux-android" ]]; then
+    # NOTE: We assume that the Android environment is set up correctly.
+    #       Make sure that `crtbegin_so.o` and `crtend_so.o` can be found.
```

Comment:
Depending on the build we'll need more here. But let's do that in a follow up when we actually have CI builds for this case as well.